TeachingRecordSystem/src/TeachingRecordSystem.Core/Jobs/EwcWalesImport/QtsImporter.cs

```diff
@@ -0,0 +1,608 @@
+        }
+        else
+        {
+            var status = results.Single(x => x.dfeta_Value == "213"); //213 - Qualified Teacher: QTS awarded in Wales
+            return (EwcWalesMatchStatus.OneMatch, status.Id);
+        }
+    }
+
+    public dfeta_classdivision? GetHEClassDivision(string classCode)
+    {
+        switch (classCode)
+        {
+            case "1":
+            case "01":
+                return dfeta_classdivision.Firstclasshonours;
+            case "2":
+            case "02":
+                return dfeta_classdivision.Uppersecondclasshonours;
+            case "3":
+            case "03":
+                return dfeta_classdivision.Lowersecondclasshonours;
+            case "4":
+            case "04":
+                return dfeta_classdivision.Undividedsecondclasshonours;
+            case "5":
+            case "05":
+                return dfeta_classdivision.Thirdclasshonours;
+            case "6":
+            case "06":
+                return dfeta_classdivision.Fourthclasshonours;
+            case "7":
+            case "07":
+                return dfeta_classdivision.Unclassifiedhonours;
+            case "8":
+            case "08":
+                return dfeta_classdivision.Aegrotat_whethertohonoursorpass;
+            case "9":
+            case "09":
+                return dfeta_classdivision.Passdegreeawardedwithouthonoursfollowinganhonourscourse;
+            case "10":
+                return dfeta_classdivision.Ordinary_includingdivisionsofordinaryifanydegreeawardedafterfollowinganonhonourscourse;
+            case "11":
+                return dfeta_classdivision.GeneralDegreedegreeawardedafterfollowinganonhonourscoursedegreethatwasnotavailabletobeclassified;
+            case "12":
+                return dfeta_classdivision.Distinction;
+            case "13":
+                return dfeta_classdivision.Merit;
+            case "14":
+                return dfeta_classdivision.Pass;
+            case "98":
+                return dfeta_classdivision.Notapplicable;
+            case "99":
+                return dfeta_classdivision.NotKnown;
+            default:
+                return null;
+        }
+    }
+
+    public async Task<(EwcWalesMatchStatus, Guid? IttQualificationId)> GetIttQualificationAsync(string ittQualificationcode)
+    {
+        var results = await cache.GetIttQualificationsAsync();
+        var qualifications = results.Where(x => x.dfeta_Value == ittQualificationcode).ToArray();
+
+        if (qualifications.Length == 0)
+            return (EwcWalesMatchStatus.NoMatch, null);
+
+        if (qualifications.Length > 1)
+            return (EwcWalesMatchStatus.MultipleMatchesFound, null);
+
+        var qualificationId = qualifications.First().Id;
+        return (EwcWalesMatchStatus.OneMatch, qualificationId);
+    }
+
+    public async Task<(EwcWalesMatchStatus, Guid? SubjectId)> FindIttSubjectAsync(string subjectCode)
```

Comment:
As with induction, all these lookups should return their own enum `Result` type.